hazelcast/include/hazelcast/client/impl/HazelcastClientInstanceImpl.h

```diff
@@ -171,14 +172,14 @@ namespace hazelcast {
                 *
                 * @param lifecycleListener Listener object
                 */
-                void addLifecycleListener(LifecycleListener *lifecycleListener);
+                boost::uuids::uuid addLifecycleListener(LifecycleListener &&lifecycleListener);
```

Comment:
Can we change this API to work similar to Java APi where we do `getLifecycleListener().addLifeCycleListener()` ? We were not able to change this before but now we can break it and change. LifecycleService also provides a few more public APIs such as `isRunning`. We can also provide this as public, they are currently on the client itself. This can be on another PR.

Reply:
Will do on a separate PR.